### PR TITLE
clear hints for comp run

### LINKED ISSUE #220: [BUG] give clear hints for user to run comp
```
➜  /Users/zhouzhengxi/Programming/golang/src/github.com/zzxwill/RudrX git:(master) bin/vela comp run -h
Error: must specify name for workload
```

Option 1:
```
➜  /Users/zhouzhengxi/Programming/golang/src/github.com/zzxwill/RudrX git:(master) bin/vela comp run -h
Error: must specify name for workload, please choose from `containerized`, `deployment`, ...
```

Opetion 2:
```
➜  /Users/zhouzhengxi/Programming/golang/src/github.com/zzxwill/RudrX git:(master) bin/vela comp run -h
Error: must specify name for workload, please run `vela workloads` to see the workload list you can choose from
```

```
➜  /Users/zhouzhengxi/Programming/golang/src/github.com/zzxwill/RudrX git:(master) vela workloads
NAME         	DEFINITION
containerized	containerizedworkloads.core.oam.dev
```

### PR DESCRIPTION
**Pull request by sunny0826**
fix #220 
/cc @zzxwill 

Signed-off-by: guoxudong <sunnydog0826@gmail.com>

### CODE REVIEW

File: pkg/cmd/comp.go [hunk 1]
```diff
@@ -90,7 +91,15 @@ func NewCompRunCommands(c types.Args, ioStreams util.IOStreams) *cobra.Command {
...
-		return "", errors.New("must specify name for workload")
+		workloadList,err := InstalledWorkloads()
+		if err != nil {
+			return "",err
+		}
+		errMsg := "can not find workload, check workloads by `vela workloads` and choose a suitable one."
+		if workloadList != "" {
+			errMsg = fmt.Sprintf("must specify name for workload ,please choose from %s.",workloadList)
+		}
+		return "", errors.New(errMsg)
```
Comment (wonderflow) [inline pkg/cmd/comp.go:102]: it's infact component name, not workload name, changing the error message is fine

Comment (sunny0826) [reply pkg/cmd/comp.go:102]: Change it to "must specify workload, please choose from 'deployment','containerized'" ?

Comment (wonderflow) [reply pkg/cmd/comp.go:102]: No, should change to "must specify name for component"

Comment (sunny0826) [reply pkg/cmd/comp.go:102]: OK

File: pkg/cmd/comp.go [hunk 2]
```diff
@@ -90,7 +91,15 @@ func NewCompRunCommands(c types.Args, ioStreams util.IOStreams) *cobra.Command {
 func GetWorkloadNameFromArgs(args []string) (string, error) {
 	argsLength := len(args)
 	if argsLength < 1 {
-		return "", errors.New("must specify name for workload")
+		workloadList,err := InstalledWorkloads()
```
Comment (wonderflow) [inline pkg/cmd/comp.go:94]: We don't need to get workload type List here.

Comment (zzxwill) [reply pkg/cmd/comp.go:94]: The list can help the user to quickly complete the command without 1) switching to `vela workloads` and choosing a suitable one, and 2) returning back to `vela comp`.

File: pkg/cmd/comp.go [hunk 3]
```diff
@@ -90,7 +91,15 @@ func NewCompRunCommands(c types.Args, ioStreams util.IOStreams) *cobra.Command {
...
+		if err != nil {
+			return "",err
+		}
+		errMsg := "can not find workload, check workloads by `vela workloads` and choose a suitable one."
+		if workloadList != "" {
+			errMsg = fmt.Sprintf("must specify name for component, please choose from %s.",workloadList)
```
Comment (zzxwill) [inline pkg/cmd/comp.go:100]:
```suggestion
			errMsg = fmt.Sprintf("must specify name for component, please choose from %s.", workloadList)
```

Comment (zzxwill) [reply pkg/cmd/comp.go:100]: Go format the changed file first:)

File: pkg/cmd/workloads.go [hunk 4]
```diff
@@ -39,3 +41,17 @@ func printWorkloadList(workloadList []types.Capability, ioStreams cmdutil.IOStre
 	ioStreams.Info(table.String())
 	return nil
 }
+
+func InstalledWorkloads() (string, error) {
```
Comment (zzxwill) [inline pkg/cmd/workloads.go:45]: I believe there is a similar function located in `pkg/plugins/local.go`. Would you please check it first?

File: pkg/cmd/comp.go (hunk 1 above)
Comment (wonderflow) [reply pkg/cmd/comp.go:105]: Maybe there's still misunderstanding here, vela comp run command is like below:

```
vela comp run -t <workload_type> component-name
```

So the lengh of arguments less than 1 just means user forgot to specify component name here, but your code is telling user to specify workload type which should be specify by `-t` parameter.

@sunny0826 @zzxwill

Comment (sunny0826) [reply pkg/cmd/comp.go:105]: I get it, my bad. Maybe I need to regroup my thoughts.

File: pkg/cmd/comp.go [hunk 5]
```diff
@@ -60,6 +60,13 @@ func NewCompRunCommands(c types.Args, ioStreams util.IOStreams) *cobra.Command {
 		Example:            "vela comp run -t <workload-type>",
 		RunE: func(cmd *cobra.Command, args []string) error {
 
+			if args[0] == "-h" {
+				err := cmd.Help()
+				if err != nil {
+					return err
+				}
+				return nil
+			}
```
Comment (sunny0826) [inline pkg/cmd/comp.go:69]: fix bug of `vela comp -h`

**Review by wonderflow — APPROVED** [review]
Good job

File: pkg/cmd/comp.go [hunk 6]
```diff
@@ -119,6 +126,21 @@ func (o *runOptions) Complete(cmd *cobra.Command, args []string, ctx context.Con
...
+		var workloadList []string
+		for _, w := range workloads {
+			workloadList = append(workloadList, w.Name)
+		}
+		errMsg := "can not find workload, check workloads by `vela workloads` and choose a suitable one."
+		if workloadList != nil {
+			errMsg = fmt.Sprintf("must specify type of workload, please use `-t` and choose from %v.", workloadList)
+		}
+		return errors.New(errMsg)
+	}
```
Comment (sunny0826) [inline pkg/cmd/comp.go:143]: clear hints for workloads

**Pull request merged**